internal/events/kafka.go

```diff
@@ -0,0 +1,115 @@
+package events
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/segmentio/kafka-go"
+	"github.com/stellar/go/support/log"
+	"golang.org/x/exp/maps"
+)
+
+type KafkaEventManager struct {
+	handlers []EventHandler
+	writer   *kafka.Writer
+	reader   *kafka.Reader
+}
+
+func NewKafkaEventManager(brokers []string, consumerTopics []string, consumerGroupID string) (*KafkaEventManager, error) {
+	k := KafkaEventManager{}
+
+	writer := kafka.NewWriter(kafka.WriterConfig{
+		Brokers:      brokers,
+		Balancer:     &kafka.RoundRobin{},
+		RequiredAcks: -1,
+	})
+
+	reader := kafka.NewReader(kafka.ReaderConfig{
+		Brokers:     brokers,
+		GroupID:     consumerGroupID,
+		GroupTopics: consumerTopics,
```

Comment:
I'm not sure about using `GroupTopics`. I never used it before so we have to look into this. 

How would consumption work ? Would we round robin between these topics and get one message from each ? In that case we may be delaying important messages in favor of less important messages. 

I think we may need to switch to an implementation that handles one topic per reader/consumer. We can have multiple handlers per consumer (e.g. `sdp.payments.payment_status_updated` would be handled both by `PatchAnchorPlatformEventHandler` and `UpdatePaymentStatusEventHandler` ).

Reply:
From the library docs:

> GroupTopics allows specifying multiple topics, but can only be used in combination with GroupID, as it is a consumer-group feature. As such, if GroupID is set, then either Topic or GroupTopics must be defined.

But, I believe we can fall into the problem you said, delaying important messages in favor of less important ones. I can change the implementation to separate the consumer from the producer, just like you [suggested here](https://github.com/stellar/stellar-disbursement-platform-backend/pull/119#discussion_r1424913197). I think this approach will make things easy to use and test.

Thank you so much for your comment!

Reply:
Changed: [87c250d](https://github.com/stellar/stellar-disbursement-platform-backend/pull/119/commits/87c250d9d3feb26893f037c7735c336056f96743)